src/middle/ssa/ssastorage.rs

```diff
@@ -225,7 +254,8 @@ impl SSAStorage {
         }
     }
 
-    fn replace_node(&mut self, i: NodeIndex, j: NodeIndex) {
+    // TODO: When i is a constant, there may be some bugs.
```

Comment:
What kind of bugs?

Reply:
Because of we maintain only one copy of a constant, if we replace a constant node, we may make other irrelevant nodes point to the replacement. 

Actually, constant node should not be replaced. Maybe I should add an assert for it?